src/types/collection.ts

```diff
@@ -465,10 +465,10 @@ export declare class SafeCollection<
    * @param options - Optional settings for the command
    * @param callback - An optional callback, a Promise will be returned if none is provided
    */
-  mapReduce<TKey = any, TValue = any>(
+  mapReduce<TKey extends ObjectId, TValue = any>(
```

Comment:
They key doesn't have to be an `ObjectId`, if you look at the examples.

https://www.mongodb.com/docs/manual/tutorial/map-reduce-examples/